batch/batches/ValidateLiveMediaServers/KAsyncValidateLiveMediaServers.class.php

```diff
@@ -22,83 +25,280 @@ public static function getType()
-    {
-        $excludeServerIds = array();
-        $serverTypesNotIn = $this->getAdditionalParams('serverTypesNotIn');
-        if ($serverTypesNotIn)
-        {
-            $serverNodes = KBatchBase::tryExecuteApiCall(array('KAsyncValidateLiveMediaServers','getExcludeServerNodesFromAPI'), array($serverTypesNotIn));
-            if ($serverNodes && $serverNodes->objects)
-            {
-                foreach($serverNodes->objects as $serverNode)
-                {
-                    $excludeServerIds[] = $serverNode->id;
-                }
-            }
-        }
-        return $excludeServerIds;
-    }
-	
 	/* (non-PHPdoc)
 	 * @see KBatchBase::run()
 	*/
 	public function run($jobs = null)
 	{
-		$entryServerNodeFilter = $this->getFilter();
-		
-		$entryServerNodePager = new KalturaFilterPager();
-		$entryServerNodePager->pageSize = 500;
-		$entryServerNodePager->pageIndex = 1;
-		
+		$this->validateLiveEntryServerNode();
+		$this->validateSimuliveEntryServerNode();
+	}
+
+	protected function validateLiveEntryServerNode()
+	{
+		$entryServerNodeFilter = $this->getLiveEntryServerNodesFilter();
+
+		$entryServerNodePager = self::getDefaultPager();
+
 		$entryServerNodes = self::$kClient->entryServerNode->listAction($entryServerNodeFilter, $entryServerNodePager);
-		
+
 		while($entryServerNodes->objects && count($entryServerNodes->objects))
 		{
 			foreach($entryServerNodes->objects as $entryServerNode)
 			{
-				try
-				{
-					/* @var $entryServerNode KalturaEntryServerNode */
-					self::impersonate($entryServerNode->partnerId);
-					self::$kClient->entryServerNode->validateRegisteredEntryServerNode($entryServerNode->id);
-					self::unimpersonate();
-				}
-				catch (KalturaException $e)
+				/* @var $entryServerNode KalturaEntryServerNode */
+				KBatchBase::tryExecuteApiCall(array('KAsyncValidateLiveMediaServers', 'validateRegisteredEntryServerNode'), array($entryServerNode->partnerId, $entryServerNode->id));
+			}
+
+			$entryServerNodePager->pageIndex++;
+			$entryServerNodes = self::$kClient->entryServerNode->listAction($entryServerNodeFilter, $entryServerNodePager);
+		}
+	}
+
+	protected function validateSimuliveEntryServerNode()
+	{
+		// getting simulive serverNode
+		$simuliveServerNode = KBatchBase::tryExecuteApiCall(array('KAsyncValidateLiveMediaServers', 'getSimuliveServerNode'), array());
+		if (!$simuliveServerNode)
+		{
+			KalturaLog::info('no simulive server node found');
+			return;
+		}
+
+		//getting all esn of the simulive serverNode
+		$entryServerNodes = KBatchBase::tryExecuteApiCall(array('KAsyncValidateLiveMediaServers', 'getEntryServerNodes'), array($simuliveServerNode->id));
+
+		// getting currently live events
+		$currentlyLiveEvents = KBatchBase::tryExecuteApiCall(array('KAsyncValidateLiveMediaServers', 'getSimulivePlayableEvents'), array());
+
+		// map between templateEntryId to sourceEntryId
+		$eventsMap = self::arrayColumn($currentlyLiveEvents, 'templateEntryId', 'sourceEntryId');
```

Comment:
as sourceEntryId can be null and you check for the null later it will work. But for readability I think we better filter simulive events only before that